### PR TITLE
fix [ bug #825 ] User first name not visible into users->perms

### PR DESCRIPTION
**Pull request by FHenry**

### CODE REVIEW

**Comment by simnandez** [comment]
Deleting a validated invoice (with a reference) may be illegal in some countries. For example in Spain must be correlative numbering without hollows, and have a list of invoices 1,2,4 for eliminating the invoice 3 entail a breach of the law.

We should warn of this in the elimination validation message?

**Comment by atm-maxime** [comment]
Dolibarr doesn't allow to delete a validated invoice if it's not the last numbered one or if it has a payment. Florian fix is just for stock regulation in case of last numbered invoice deletion.

**Comment by FHenry** [comment]
But i've you want to add warning message about deleting invoce can can have legal issue, it's up to you. Justt add it in translation files.

File: htdocs/langs/es_ES/agenda.lang [hunk 1]
```diff
@@ -41,6 +41,7 @@ ActionsEvents=Eventos para que Dolibarr cree una acción de forma automática
 PropalValidatedInDolibarr=Presupuesto %s validado
 InvoiceValidatedInDolibarr=Factura %s validada
 InvoiceBackToDraftInDolibarr=Factura %s devuelta a borrador
+InvoiceDeleteDolibarr=Factura %s elimina
```
Comment (marcosgdf) [inline htdocs/langs/es_ES/agenda.lang]: Factura %s eliminada

File: htdocs/langs/en_US/stocks.lang [hunk 2]
```diff
@@ -55,6 +55,7 @@ DeStockOnShipment=Decrease real stocks on shipment validation
 ReStockOnBill=Increase real stocks on suppliers invoices/credit notes validation
 ReStockOnValidateOrder=Increase real stocks on suppliers orders approbation
 ReStockOnDispatchOrder=Increase real stocks on manual dispatching into warehouses, after supplier order receiving
+ReStockOnDispatchOrder=Increase real stocks on invocie deletion
```
Comment (marcosgdf) [inline htdocs/langs/en_US/stocks.lang]: invoice*

File: htdocs/langs/es_ES/stocks.lang [hunk 3]
```diff
@@ -54,6 +54,7 @@ DeStockOnShipment=Decrementar los stocks físicos sobre los envíos
 ReStockOnBill=Incrementar los stocks físicos sobre las facturas/abonos de proveedores
 ReStockOnValidateOrder=Incrementar los stocks físicos sobre los pedidos a proveedores
 ReStockOnDispatchOrder=Incrementa los stocks físicos en el desglose manual de la recepción de los pedidos a proveedores en los almacenes
+ReStockOnDeleteInvoice=Incrementa los stocks físicos a la suppresion de las facturas
```
Comment (marcosgdf) [inline htdocs/langs/es_ES/stocks.lang]: Incrementa los stocks físicos en la eliminación de facturas

**Comment by FHenry** [comment]
Ok, my spanish is really bad... I do the translation correction.

**Comment by eldy** [comment]
I don't think "increase stock" on delete invoice is not a valid option for user :
List of options to increase stocks are a choice user want to have to increase stock in a "standard" workflow (when everything is correct). Deleting an invoice is not the standard workflow so it must not appears as a standard option for increasing a stock.
Moreover, deleting can be done only onto last invoice, just to solve a problem when user makes en error by entering its invoice. Deleteing in this case means it is a work around to manually solve a situation. Fixing manualy thing, means we can also fix manually stocks (because we should very rarely delete an invoice).
Another reason is deleting an invoice does not necessarly means sotck has change. Product can be sent or lost, we don't want invoice because customer does no more exists, we

So increase stock on delete it must not depends on an increase option (it is not an workflow choice) but depends on:
- decrease option is on validate invoice (This already existing option IS the workflow choice that means we decreased when creating so we should increase when deleting).
- and we msut also increase if a user click onto a checkbox durign the delete confirm message to ask
  "Delete also stocks that were increased ?".  And this checkbox must be visible only if the standard option "decrease on validate invoice" were on.

**Comment by FHenry** [comment]
Ok let's do that during DevCamp. J'ai hate !

**Pull request merged**